python/pyspark/pandas/indexing.py

```diff
@@ -174,7 +174,7 @@ def __getitem__(self, key) -> Union["Series", "DataFrame", Scalar]:
         if len(pdf) < 1:
             raise KeyError(name_like_string(row_sel))
 
-        values = pdf.iloc[:, 0].values
+        values = pdf.iloc[:, 0].values  # type: ignore
```

Comment:
I guess `values = cast(pd.DataFrame, pdf).iloc[:, 0].values`?
or we should cast right after `toPandas()` above?